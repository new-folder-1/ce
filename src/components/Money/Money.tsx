import * as React from 'react';

import { Currency } from '../../types';

export interface MoneyProps {
    amount: number;
    currency: Currency;
}

export const Money = React.memo(({ amount, currency }: MoneyProps) => (
    <span>
        {amount.toLocaleString('en', {
            style: 'currency',
            currency: currency
        })}
    </span>
));
