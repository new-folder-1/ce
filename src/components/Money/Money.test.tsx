import React from 'react';
import { shallow } from 'enzyme';

import { Money } from './Money';

describe('Money', () => {
    it('snapshot renders', () => {
        expect(
            shallow(<Money amount={1} currency="USD" />)
        ).toMatchSnapshot();
    })

    it('renders with currency symbol', () => {
        const component = shallow(<Money amount={20.01} currency="GBP" />)
        expect(component.text()).toEqual('Â£20.01');
    });

    it('renders with thousands separator', () => {
        const component = shallow(<Money amount={1000} currency="USD" />)
        expect(component.text()).toEqual('$1,000.00');
    });
});
