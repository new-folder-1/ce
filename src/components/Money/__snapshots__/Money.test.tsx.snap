// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Money snapshot renders 1`] = `
<span>
  $1.00
</span>
`;
